lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -255,120 +259,49 @@ data ApiErrorCode
---
---     > "name *-*"
---
-data Iso8601Range (name :: Symbol)
-    = Iso8601Range (Maybe UTCTime) (Maybe UTCTime)
-    deriving (Eq, Generic, Show)
+newtype Iso8601Time = Iso8601Time
+    { getIso8601Time :: UTCTime
+    } deriving (Eq, Ord, Show)
 
-instance KnownSymbol name => ToText (Iso8601Range name) where
-    toText (Iso8601Range t1 t2) = prefix <> " " <> suffix
-      where
-        prefix = T.pack $ symbolVal (Proxy @name)
-        suffix = timeToText t1 <> "-" <> timeToText t2
-        timeToText = \case
-            Nothing -> "*"
-            Just t -> utcTimeToText iso8601BasicUtc t
-
-instance KnownSymbol name => FromText (Iso8601Range name) where
-    fromText t = do
-        (prefix, suffix) <- guardM (splitAtLastOccurrence ' ' t)
-            "Unable to find required space separator character."
-        (t1, t2) <- guardM (splitAtLastOccurrence '-' suffix)
-            "Unable to find required hyphen separator character."
-        guardB (prefix == expectedPrefix) $
-            "Invalid prefix string found. Expecting: " <> show expectedPrefix
-        v1 <- guardM (parseTime t1) $
-            "Invalid start time string: " <> show t1
-        v2 <- guardM (parseTime t2) $
-            "Invalid end time string: " <> show t2
-        pure $ Iso8601Range v1 v2
+instance ToText Iso8601Time where
+    toText = utcTimeToText iso8601ExtendedUtc . getIso8601Time
+
+instance FromText Iso8601Time where
+    fromText t =
+        Iso8601Time <$> maybeToEither err (utcTimeFromText iso8601 t)
       where
-        expectedPrefix = T.pack $ symbolVal (Proxy @name)
+        err = TextDecodingError $ mempty
+            <> "Unable to parse time argument: '"
+            <> T.unpack t
+            <> "'. Expecting ISO 8601 format (basic or extended)."
+
+instance FromHttpApiData Iso8601Time where
+    parseUrlPiece = first (T.pack . getTextDecodingError) . fromText
 
-        guardB :: Bool -> String -> Either TextDecodingError ()
-        guardB b err = if b then Right () else raise err
+instance ToHttpApiData Iso8601Time where
+    toUrlPiece = toText
 
-        guardM :: Maybe a -> String -> Either TextDecodingError a
-        guardM ma err = maybe (raise err) Right ma
+-- | Represents a sort order, applicable to a query.
+data SortOrder
+    = Ascending
+        -- ^ Sort in ascending order.
+    | Descending
+        -- ^ Sort in descending order.
+    deriving (Bounded, Enum, Eq, Generic, Show)
 
-        raise :: forall a . String -> Either TextDecodingError a
-        raise err = Left $ TextDecodingError $
-            "Error encountered while decoding ISO 8601 time range: " <> err
+instance ToText SortOrder where
+    toText = toTextFromBoundedEnum CamelCase
 
-        parseTime :: Text -> Maybe (Maybe UTCTime)
-        parseTime = \case
-            "*" -> pure Nothing
-            timeText -> pure <$> utcTimeFromText [iso8601BasicUtc] timeText
+instance FromText SortOrder where
+    fromText = fromTextToBoundedEnum CamelCase
```

Comment:
mostly everything in our API is `snake_cased` isn't it? It won't matter much here since these are single-word constructors, but for consistency :grimacing: ...

Reply:
Fair enough! I'll update this.

Reply:
@KtorZ Fixed.